torch/csrc/tensor/python_tensor.cpp

```diff
@@ -392,6 +392,9 @@ Device getDevice(const at::Tensor& tensor) {
   if (tensor.is_cuda()) {
     return at::Device(at::DeviceType::CUDA, tensor.get_device());
   }
+  if (tensor.type_id() == XLATensorId() || tensor.type_id() == MSNPUTensorId()) {
+    return at::Device(computeDeviceType(tensor.type_id()), tensor.get_device());
```

Comment:
I feel like maybe we can remove this getDevice function entirely and just replace occurrences with a call to tensor.device().

Reply:
Done, I've only found two call sites.